packages/core/src/extensions/Blocks/nodes/Block.ts

```diff
@@ -162,275 +109,408 @@ export const Block = Node.create<IBlock>({
+            state.schema.node(type).type,
+            attributes
+          );
+
+          return true;
+        },
+      // Changes the block at a given position to a given content type if it's empty, otherwise creates a new block of
+      // that type below it.
+      BNCreateBlockOrSetContentType:
+        (posInBlock, type, attributes) =>
+        ({ state, chain }) => {
+          const blockInfo = getBlockInfoFromPos(state.doc, posInBlock);
+          if (blockInfo === undefined) {
+            return false;
+          }
+
+          const { node, startPos, endPos } = blockInfo;
+
+          if (node.textContent.length === 0) {
+            const oldBlockContentPos = startPos + 1;
+
+            return chain()
+              .BNSetContentType(posInBlock, type, attributes)
+              .setTextSelection(oldBlockContentPos)
+              .run();
+          } else {
+            const newBlockInsertionPos = endPos + 1;
+            const newBlockContentPos = newBlockInsertionPos + 1;
+
+            return chain()
+              .BNCreateBlock(newBlockInsertionPos)
+              .BNSetContentType(newBlockContentPos, type, attributes)
+              .setTextSelection(newBlockContentPos)
+              .run();
           }
-          return false;
         },
-      joinBackward:
-        () =>
-        ({ view, dispatch, state }) =>
-          joinBackward(state, dispatch, view), // Override default joinBackward with edited command
     };
   },
+
   addProseMirrorPlugins() {
-    return [PreviousBlockTypePlugin(), OrderedListPlugin()];
+    return [PreviousBlockTypePlugin()];
   },
+
   addKeyboardShortcuts() {
     // handleBackspace is partially adapted from https://github.com/ueberdosis/tiptap/blob/ed56337470efb4fd277128ab7ef792b37cfae992/packages/core/src/extensions/keymap.ts
     const handleBackspace = () =>
       this.editor.commands.first(({ commands }) => [
-        // Maybe the user wants to undo an auto formatting input rule (e.g.: - or #, and then hit backspace) (source: tiptap)
+        // Deletes the selection if it's not empty.
+        () => commands.deleteSelection(),
+        // Undoes an input rule if one was triggered in the last editor state change.
         () => commands.undoInputRule(),
-        // maybe convert first text block node to default node (source: tiptap)
+        // Removes a level of nesting if the block is indented and the block is empty.
         () =>
-          commands.command(({ tr }) => {
-            const { selection, doc } = tr;
-            const { empty, $anchor } = selection;
-            const { pos, parent } = $anchor;
-            const isAtStart = Selection.atStart(doc).from === pos;
-
-            if (
-              !empty ||
-              !isAtStart ||
-              !parent.type.isTextblock ||
-              parent.textContent.length
-            ) {
-              return false;
+          commands.command(({ state }) => {
+            const { depth } = getBlockInfoFromPos(
```

Comment:
should we get doc and selection from `state` or from `tr`? My guess is the latter, just how it was before

Reply:
I think it comes from `state` so probably it doesn't really matter: https://github.com/ueberdosis/tiptap/blob/main/packages/core/src/CommandManager.ts